File: src/website/services/userMgmt/setProfileImage.ts
import CanisterClientFactory from "../CanisterClientFactory";

export default async function(imageId: string) : Promise<SetProfileImageResponse> {
    const client = CanisterClientFactory.current!.userMgmtClient;
    const response = await client.set_profile_image(imageId);

    if ("Success" in response) {
        return SetProfileImageResponse.Success;
    } else if ("UserNotFound" in response) {
        return SetProfileImageResponse.UserNotFound;
    } else {
        throw new Error("Unrecognised 'set_profile_image' response");
    }
}

export enum SetProfileImageResponse {
    Success,
    UserNotFound
}

File: src/services/user_mgmt/src/updates/set_profile_image.rs
use self::Response::*;
use crate::domain::user_store::UserStore;
use ic_cdk::export::candid::CandidType;
use ic_cdk::storage;
use shared::timestamp;

pub fn update(image_id: String) -> Response {
    let me = shared::user_id::get_current();
    let now = timestamp::now();

    let user_store: &mut UserStore = storage::get_mut();

    match user_store.set_profile_image(&me, image_id, now) {
        false => UserNotFound,
        true => Success,
    }
}

#[derive(CandidType)]
pub enum Response {
    Success,
    UserNotFound,
}
